src/main/java/com/arpnetworking/tsdcore/statistics/TPStatistic.java

```diff
@@ -93,13 +92,7 @@ public Quantity calculate(final List<Quantity> orderedValues) {
      */
     @Override
     public Quantity calculateAggregations(final List<AggregatedData> aggregations) {
-        final List<Quantity> allSamples = Lists.newArrayList();
-        for (final AggregatedData aggregation : aggregations) {
-            allSamples.addAll(aggregation.getSamples());
-        }
-        Collections.sort(allSamples);
-        final int index = (int) (Math.ceil((_percentile / 100.0) * (allSamples.size() - 1)));
-        return allSamples.get(index);
+        throw new UnsupportedOperationException("Unsupported operation: calculateAggregations(List<AggregatedData>)");
```

Comment:
I didn't want to go down any unnecessary refactoring code paths in a single change. This already included clean-up for AggregatedData and PeriodicData which as you can tell was quite a sprawl. The statistics also need to be extracted into a shared library so I didn't want the copy in MAD to diverge too much from that in CAgg.